Source/EditorTools/BaseTool.cs

```diff
@@ -1,46 +1,42 @@
 #if UNITY_EDITOR
 
+using System;
+using System.Collections.Generic;
+using System.Linq;
 using iffnsStuff.MarchingCubeEditor.Core;
+using iffnsStuff.MarchingCubeEditor.SceneEditor;
 using UnityEditor;
 using UnityEngine;
 
-public abstract class BaseTool
+public class BaseTool
 {
-    protected MarchingCubesController linkedMarchingCubesController;
-
-    public abstract string displayName { get; }
-
-    public BaseTool(MarchingCubesController linkedMarchingCubesController)
+    protected MarchingCubeEditor Editor { get; private set; }
+    protected MarchingCubesController Controller => Editor.Controller;
```

Comment:
I usually write `linkedMarchingCubesController` for example with the following thoughts:
- Lower case for non-public variables.
- Linked to indicate that this is an external reference that needs to be set up correctly and could potentially be broken if not. Likely caused by the many times I forgot to assign a reference in the Unity inspector.
- Type to indicate what type of element it is.

Reply:
I made them PascalCase mainly because of VSCode. I usually indicate constants by making them uppercase, but we ofc can also make it camelCase. I did not include the variable type simple out of convenience.
![image](https://github.com/user-attachments/assets/0af85aad-24b2-4898-9225-cb5e01ddea2a)

> Linked to indicate that this is an external reference that needs to be set up correctly and could potentially be broken if not.

Is handled by the [loading script](https://github.com/danielfvm/MarchingCubeEditorForUnity/blob/538659dfa75c77390e0f4b1f86f20cf86aeed791/Source/EditorTools/BaseTool.cs#L37), so should never be broken or null.

Reply:
I agree with CamelCase.
But I would still call them `LinkedMarchingCubeEditor` / `LinkedMarchingCubeController` since its part of the base class and there might also be different editor and controllers for other applications.

Reply:
Okay I renamed it.